src/mysky/encrypted_files.ts

```diff
@@ -196,15 +196,19 @@ export function deriveEncryptedFileTweak(pathSeed: string): string {
  * @param subPath - The path.
  * @param isDirectory - Whether the path is a directory.
  * @returns - The path seed for the given path.
+ * @throws - Will throw if the input sub path is not a valid path.
  */
 export function deriveEncryptedFileSeed(pathSeed: string, subPath: string, isDirectory: boolean): string {
   validateHexString("pathSeed", pathSeed, "parameter");
   validateString("subPath", subPath, "parameter");
   validateBoolean("isDirectory", isDirectory, "parameter");
 
   let pathSeedBytes = hexToUint8Array(pathSeed);
-  subPath = sanitizePath(subPath);
-  const names = subPath.split("/");
+  const sanitizedPath = sanitizePath(subPath);
+  if (sanitizedPath === null) {
```

Comment:
strict null comparisons on strings always make me cringe a bit. That really only works well if you can be absolutely certain the validator functions properly. I think in this case it doesn't handle the empty string case for instance.

Reply:
I don't really follow you. We want to check specifically for `null` from `sanitizePath` because it indicates an invalid path.